redhat-appstudio/service-provider-integration-operator
# introduce spi instance id config option and prefix aws secrets with it

### What does this PR do?
new cli option `--spi-instance-id` or env `SPIINSTANCEID`. For now, it is used only for AWS secrets.

### Screenshot/screencast of this PR



### What issues does this PR fix or reference?

https://issues.redhat.com/browse/SVPI-367

### How to test this PR?


```
quay.io/redhat-appstudio/pull-request-builds:spi-controller-#
quay.io/redhat-appstudio/pull-request-builds:spi-oauth-#
```

1. Deploy with AWS secret storage
2. set `SPIINSTANCEID` at `spi-system/spi-shared-environment-config` to some value
3. create binding and provide the token
4. see secret is stored prefixed with `SPIINSTANCEID`


![Screenshot 2023-03-20 at 13 14 01](https://user-images.githubusercontent.com/1160903/226336509-8a767c30-dc64-4a19-8bb0-7b26ec137da0.png)
![Screenshot 2023-03-20 at 13 15 16](https://user-images.githubusercontent.com/1160903/226336536-99e19a1b-b90d-4df7-91f5-21f64a50ad50.png)
![Screenshot 2023-03-20 at 13 15 23](https://user-images.githubusercontent.com/1160903/226336547-f1adc434-444f-45c9-bd13-7bc12701cb5f.png)
